code/game/objects/structures/cannons/mounted_guns/mounted_gun.dm

```diff
@@ -0,0 +1,187 @@
+	density = TRUE
+	anchored = FALSE
+	icon = 'icons/obj/weapons/cannons.dmi'
+	icon_state = "falconet_patina"
+	var/icon_state_base = "falconet_patina"
+	var/icon_state_fire = "falconet_patina_fire"
+	max_integrity = 300
+	///whether the cannon can be unwrenched from the ground. Anchorable_cannon equivalent.
+	var/anchorable_gun = TRUE
+	//Max shots per firing of the gun.
+	var/max_shots_per_fire = 1
+	//Shots currently loaded. Should never be more than max_shots_per_fire.
+	var/shots_in_gun = 1
+	//shots added to gun, per piece of ammo loaded.
+	var/shots_per_load = 1
+	//Accepted "ammo" type
+	var/obj/item/ammo_type = /obj/item/ammo_casing/strilka310
+	//Projectile from said gun. Doesnt automatically inherit said ammo's projectile in case you wanted to make a gun that shoots floor tiles or something.
+	var/obj/item/projectile_type = /obj/projectile/bullet/strilka310
+	//If the gun has anything in it.
+	var/loaded_gun = TRUE
+	//If the gun is currently loaded with its maximum capacity.
+	var/fully_loaded_gun = TRUE
+	//delay in firing the gun after lighting
+	var/fire_delay = 5
+	//Delay between shots
+	var/shot_delay = 3
+	//If the gun shakes the camera when firing
+	var/firing_shakes_camera = TRUE
+	//sound of firing for all but last shot
+	var/fire_sound = 'sound/weapons/gun/general/mountedgun.ogg'
+	//sound of firing for last shot
+	var/last_fire_sound = 'sound/weapons/gun/general/mountedgunend.ogg'
+
+/obj/structure/mounted_gun/wrench_act(mob/living/user, obj/item/tool)
+	. = ..()
+	if(!anchorable_gun) // Can't anchor an unanchorable gun.
+		return FALSE
+	default_unfasten_wrench(user, tool)
+	return ITEM_INTERACT_SUCCESS
+
+//Covers Reloading and lighting of the gun
+/obj/structure/mounted_gun/attackby(obj/item/ammo_casing/used_item, mob/user, params)
+	var/ignition_message = used_item.ignition_effect(src, user) // Checks if item used can ignite stuff.
+	if(istype(used_item, ammo_type))
+		if(fully_loaded_gun)
+			balloon_alert(user, "already fully loaded!")
+			return
+		else
+			shots_in_gun = shots_in_gun + shots_per_load //Add one to the shots in the gun
+
+			loaded_gun = TRUE // Make sure it registers theres ammo in there, so it can fire.
+			QDEL_NULL(used_item)
+			if(shots_in_gun >= max_shots_per_fire)
+				shots_in_gun = max_shots_per_fire // in case of somehow firing only some of a guns shots, and reloading, you still cant get above the maximum ammo size.
+				fully_loaded_gun = TRUE //So you cant load extra.
+			return
+
+	else if(ignition_message) // if item the player used ignites, light the gun!
+		visible_message(ignition_message)
+		user.log_message("fired a cannon", LOG_ATTACK)
+		log_game("[key_name(user)] fired a cannon in [AREACOORD(src)]")
+		addtimer(CALLBACK(src, PROC_REF(fire)), fire_delay) //uses fire proc as shown below to shoot the gun
+		return
+	..()
+
+/obj/structure/mounted_gun/proc/fire()
+	if (!loaded_gun)
+		balloon_alert_to_viewers("Gun is not loaded!","",2)
```

Comment:
```suggestion
		balloon_alert_to_viewers("gun is not loaded!", vision_distance = 2)
```

balloon alerts should be lower-cased in general

Reply:
ah ok, didnt know!